app/views/backend/trial_balances/show.html.haml

```diff
@@ -31,44 +34,65 @@
         %tr
           %th{rowspan: 2}= JournalEntryItem.human_attribute_name(:account_number)
           %th{rowspan: 2}= JournalEntryItem.human_attribute_name(:account_name)
-          %th{colspan: 2}= :total.tl
-          %th{colspan: 2}= :balance.tl
+          %th.center{colspan: 4}= :total.tl
+          %th.center{colspan: 4}= :balance.tl
         %tr
-          %th= JournalEntry.human_attribute_name(:debit)
-          %th= JournalEntry.human_attribute_name(:credit)
-          %th= JournalEntry.human_attribute_name(:debit)
-          %th= JournalEntry.human_attribute_name(:credit)
-      - for item in @balance
+          %th.right= JournalEntry.human_attribute_name(:debit)
+          %th.right= JournalEntry.human_attribute_name(:credit)
+          %th.right= JournalEntry.human_attribute_name(:debit) + " N-1"
+          %th.right= JournalEntry.human_attribute_name(:credit) + " N-1"
+          %th.right= JournalEntry.human_attribute_name(:debit)
+          %th.right= JournalEntry.human_attribute_name(:credit)
+          %th.right= JournalEntry.human_attribute_name(:debit) + " N-1"
+          %th.right= JournalEntry.human_attribute_name(:credit) + " N-1"
+      - @balance.each_with_index do |item, index|
         - d = item[0].size-1
+        - prev_item = @prev_balance[index] || []
         - if item[1].to_i > 0
           - account = Account.find(item[1])
           %tr{class: cycle(:odd, :even)}
             %td.code= link_to(account.number, action: :show, controller: :accounts, id: account.id)
             %td= link_to(account.name, action: :show, controller: :accounts, id: account.id)
             %td.dec= number_to_accountancy(item[2], currency)
             %td.dec= number_to_accountancy(item[3], currency)
+            %td.dec= number_to_accountancy(prev_item[2], currency)
+            %td.dec= number_to_accountancy(prev_item[3], currency)
             %td.dec= number_to_accountancy(item[4], currency) if item[4].to_f > 0
             %td.dec= number_to_accountancy(-item[4].to_f, currency) if item[4].to_f < 0
+            %td.dec= number_to_accountancy(prev_item[4], currency) if prev_item[4].to_f > 0
```

Comment:
No `to_f` here? What's the difference with the next line?

Reply:
Good question, to be honest I just copy/pasted
`%td.dec= number_to_accountancy(item[4], currency) if item[4].to_f > 0`
`%td.dec= number_to_accountancy(-item[4].to_f, currency) if item[4].to_f < 0`